### PR TITLE
Fly scans

### PR DESCRIPTION
**Pull request by mrakitin**
This PR implements a flyer to perform detector triggering based on motor position callback reported to EPICS.

The developed flyer can be used as a part of the plan which would perform a 2-D scan with a step motor in the "slow" dimension and a flying motor in the "fast" dimension. Dark frames can be collected as a part of the plan.

### CODE REVIEW

File: startup/81_xpd_map_flyscan.py [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+    yield from bps.stage(detector)
+
+
+import itertools
+from collections import deque
+from pathlib import Path
+from event_model import compose_resource
+
+
+class XPDFlyer:
```
Comment (mrakitin) [inline startup/81_xpd_map_flyscan.py:267]: The flyer starts here. Above is the old code.

File: startup/81_xpd_map_flyscan.py [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+
+    # Restage.
+    yield from bps.unstage(detector)
+    yield from bps.stage(detector)
+
+
+import itertools
+from collections import deque
+from pathlib import Path
+from event_model import compose_resource
```
Comment (mrakitin) [inline startup/81_xpd_map_flyscan.py:264]: Those imports need to be moved to the top of the file.

**Review by mrakitin — COMMENTED** [review]
Submitting a couple of notes for reviewers.

File: startup/81_xpd_map_flyscan.py [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+        self.__snapshot = None
+
+    def set_snaphsot(self, snapshot):
+        self.__snapshot = snapshot
+
+    def __getattr__(self, key):
+        return getattr(self.__snapshot, key)
+
+
+def _extarct_motor_pos(mtr):
```
Comment (AbbyGi) [inline startup/81_xpd_map_flyscan.py:38]: This looks like a typo to me.
```suggestion
def _extract_motor_pos(mtr):
```

File: startup/81_xpd_map_flyscan.py [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+            yield from bps.abs_set(fly_motor, _fly_stop + _backoff, group=fly_group)
+            # TODO gate starting to take data on motor position
+            for j in range(fly_pixels):
+                print(time.time())
+                fly_pixel_group = short_uid("fly_pixel")
+                for d in dets:
+                    yield from bps.trigger(d, group=fly_pixel_group)
+
+                # grab motor position right after we trigger
+                start_pos = yield from _extarct_motor_pos(fly_motor)
```
Comment (AbbyGi) [inline startup/81_xpd_map_flyscan.py:208]: Typo here, too.
```suggestion
                start_pos = yield from _extract_motor_pos(fly_motor)
```

File: startup/81_xpd_map_flyscan.py [hunk 5]
```diff
@@ -0,0 +1,472 @@
...
+                yield from bps.mv(px_start, start_pos)
+                # wait for frame to finish
+                yield from bps.wait(group=fly_pixel_group)
+
+                # grab the motor position
+                stop_pos = yield from _extarct_motor_pos(fly_motor)
```
Comment (AbbyGi) [inline startup/81_xpd_map_flyscan.py:214]: Typo here as well.
```suggestion
                stop_pos = yield from _extract_motor_pos(fly_motor)
```

**Review by AbbyGi — COMMENTED** [review]
Looks good. Just some minor typos.

**Review by AbbyGi — APPROVED** [review]

**Pull request merged**